cpp/oneapi/dal/algo/triangle_counting/test/triangle_counting_test.cpp

```diff
@@ -0,0 +1,394 @@
+    }
+
+    template <typename GraphType>
+    void check_local_and_global_task() {
+        GraphType graph_data;
+        const auto graph = create_graph<GraphType>();
+        std::int64_t vertex_count = graph_data.get_correct_vertex_count();
+        std::int64_t global_triangle_count = graph_data.get_correct_triangle_count();
+
+        std::allocator<char> alloc;
+        const auto tc_desc = dal::preview::triangle_counting::descriptor<
+            float,
+            dal::preview::triangle_counting::method::ordered_count,
+            dal::preview::triangle_counting::task::local_and_global,
+            std::allocator<char>>(alloc);
+
+        const auto result_vertex_ranking = dal::preview::vertex_ranking(tc_desc, graph);
+
+        auto local_triangles_table = result_vertex_ranking.get_ranks();
+        const auto &local_triangles =
+            static_cast<const dal::homogen_table &>(local_triangles_table);
+        const auto local_triangles_data = local_triangles.get_data<std::int64_t>();
+
+        REQUIRE(result_vertex_ranking.get_global_rank() == global_triangle_count);
+        REQUIRE(local_triangles_table.get_row_count() == vertex_count);
+
+        int correct_local_triangle_count = 0;
+        for (std::int64_t i = 0; i < vertex_count; i++) {
+            if (local_triangles_data[i] == graph_data.local_triangles[i]) {
+                correct_local_triangle_count++;
+            }
+        }
+        REQUIRE(correct_local_triangle_count == vertex_count);
+    }
+
+    template <typename GraphType>
+    void check_global_task_relabeled() {
+        GraphType graph_data;
+        const auto graph = create_graph<GraphType>();
+        std::int64_t global_triangle_count = graph_data.get_correct_triangle_count();
+
+        std::allocator<char> alloc;
+        auto tc_desc = dal::preview::triangle_counting::descriptor<
+            float,
+            dal::preview::triangle_counting::method::ordered_count,
+            dal::preview::triangle_counting::task::global,
+            std::allocator<char>>(alloc);
+
+        const auto relabel = tc_desc.get_relabel();
+        if (relabel == dal::preview::triangle_counting::relabel::yes) {
+            const auto result_vertex_ranking = dal::preview::vertex_ranking(tc_desc, graph);
+            REQUIRE(result_vertex_ranking.get_global_rank() == global_triangle_count);
+        }
+    }
+
+    template <typename GraphType>
+    void check_global_task_not_relabeled() {
+        GraphType graph_data;
+        const auto graph = create_graph<GraphType>();
+        std::int64_t global_triangle_count = graph_data.get_correct_triangle_count();
+
+        std::allocator<char> alloc;
+        auto tc_desc = dal::preview::triangle_counting::descriptor<
+            float,
+            dal::preview::triangle_counting::method::ordered_count,
+            dal::preview::triangle_counting::task::global,
+            std::allocator<char>>(alloc);
```

Comment:
set descriptor option relabel::no to be sure.

Reply:
It is done on line 338. Is it incorrect?

Reply:
line 338 seems to be outdated. API of TC: auto desc = descriptor<>().set_relabel(triangle_counting::relabel::no);
Ok, It is fixed below.